### PR TITLE
Add support for liveness probe check

### PR DESCRIPTION
**Pull request by adamkingit**

### CODE REVIEW

File: lib/EventHandler.js [hunk 1]
```diff
@@ -23,6 +24,13 @@ module.exports = class EventHandler {
     this._kubeResourceMeta = params.kubeResourceMeta;
     this._kc = params.kubeClass;
     this._factory = params.factory;
+    if (params._livenessInterval) {
```
Comment (alewitt2) [inline lib/EventHandler.js]: non-necessary existence check

File: lib/EventHandler.js [hunk 2]
```diff
@@ -23,6 +24,13 @@ module.exports = class EventHandler {
     this._kubeResourceMeta = params.kubeResourceMeta;
     this._kc = params.kubeClass;
     this._factory = params.factory;
+    if (params.livenessInterval) {
```
Comment (alewitt2) [inline lib/EventHandler.js]: non necessary existence check

File: lib/EventHandler.js [hunk 3]
```diff
@@ -47,6 +55,12 @@ module.exports = class EventHandler {
     };
     this._wm = new watchman(opt, (data) => this.eventHandler(data));
     this._wm.watch();
+    if(this._livenessInterval) {
```
Comment (alewitt2) [inline lib/EventHandler.js]: `this._livenessInterval && this._wm.watching`

File: lib/EventHandler.js [hunk 4]
```diff
@@ -47,6 +55,12 @@ module.exports = class EventHandler {
...
+      touch.sync('/tmp/liveness');
+      setInterval(() => {
+        touch.sync('/tmp/liveness');
```
Comment (alewitt2) [inline lib/EventHandler.js]: if(this._wm.watching){
  touch.sync('/tmp/liveness');
}

File: lib/EventHandler.js (hunk 2 above)
Comment (adamkingit) [reply lib/EventHandler.js]: Not necessary but avoids constructing a string to be garbage collected later when its not set

File: lib/EventHandler.js (hunk 4 above)
Comment (adamkingit) [reply lib/EventHandler.js]: done

**Pull request merged**